### PR TITLE
plugins: vision: use complete visual odometry msg to send to the FCU

### PR DESCRIPTION
**Pull request by TSC21**
Purpose: add a more complete simulated odometry data sent from Gazebo to the FCU.

### CODE REVIEW

**Comment by TSC21** [comment]
Well looks like CMake restarted checking for the ROS Mavlink version first again somehow without a visible reason (Seems the latest release is not available by apt-get yet). Have to check why.

File: src/gazebo_vision_plugin.cpp [hunk 1]
```diff
@@ -167,12 +194,46 @@ void VisionPlugin::OnUpdate(const common::UpdateInfo&)
...
-    odom_msg.set_y(pose_model.Pos().Y() + noise.Y() + _bias.Y());
-    odom_msg.set_z(pose_model.Pos().Z() + noise.Z() + _bias.Y());
-    odom_msg.set_roll(pose_model.Rot().Roll());
-    odom_msg.set_pitch(pose_model.Rot().Pitch());
-    odom_msg.set_yaw(pose_model.Rot().Yaw());
+
+    gazebo::msgs::Vector3d* position = new gazebo::msgs::Vector3d();
+    position->set_x(pose_model.Pos().X() + noise_pos.X() + _bias.X());
+    position->set_x(pose_model.Pos().Y() + noise_pos.Y() + _bias.Y());
+    position->set_x(pose_model.Pos().Z() + noise_pos.Z() + _bias.Z());
```
Comment (ChristophTobler) [inline src/gazebo_vision_plugin.cpp]: `set_x()` -> `set_y()` and `set_z()`

**Review by ChristophTobler — CHANGES_REQUESTED** [review]
This doesn't seem to be tested... The gazebo error is resolved by the comment but still, if I run this PR with `EKF2_AID_MASK = 24` and `EKF2_HGT_MODE = 3` the vehicle immediately flies away after takeoff.

File: src/gazebo_vision_plugin.cpp (hunk 1 above)
Comment (TSC21) [reply src/gazebo_vision_plugin.cpp]: Yeah. Typo. Thanks.

**Comment by TSC21** [comment]
> This doesn't seem to be tested... The gazebo error is resolved by the comment but still, if I run this PR with `EKF2_AID_MASK = 24` and `EKF2_HGT_MODE = 3` the vehicle immediately flies away after takeoff.

Yes this still needs some extensive testing. Thanks for giving it a try @ChristophTobler. I will update you when I get this to a working stage.

**Pull request closed without merging**